plugins/modules/openssh_cert.py

```diff
@@ -273,6 +283,22 @@ def generate(self, module):
             if self.pkcs11_provider:
                 args.extend(['-D', self.pkcs11_provider])
 
+            if self.use_agent:
+                proc = module.run_command([self.ssh, '-Vq', self.path])
```

Comment:
I think it's better to do the version check in `main()`, and fail there, before any other logic is executed. Then the behavior will be a lot more consistent, and it will always fail if `use_agent` is specified and the version is too old, instead of only failing in case the certificate needs to be (re-)generated.

Reply:
I moved the check to main as early as I could.